compute/main.go

```diff
@@ -0,0 +1,656 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/compute/armcompute"
+	"github.com/Azure/azure-sdk-for-go/sdk/network/armnetwork"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/armcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/resources/armresources"
+	"github.com/Azure/azure-sdk-for-go/sdk/to"
+)
+
+var (
+	subscriptionId string
+	location       = "westus2"
+)
+
+// Instance
+var (
+	resourceGroupName = "sample-resource-group"
+	vmName            = "sample-vm"
+	vnetName          = "sample-vnet"
+	subnetName        = "sample-subnet"
+	nsgName           = "sample-nsg"
+	nicName           = "sample-nic"
+	diskName          = "sample-disk"
+	publicIPName      = "sample-public-ip"
+)
+
+func main() {
+	subscriptionId = os.Getenv("AZURE_SUBSCRIPTION_ID")
+	if len(subscriptionId) == 0 {
+		log.Println("AZURE_SUBSCRIPTION_ID is not set.")
+		return
+	}
+	log.Println("AZURE_SUBSCRIPTION_ID:", subscriptionId)
+	//create virtual machine
+	createVM()
+
+	//delete virtual machine
+	// deleteVM()
```

Comment:
We could add an environment variable to control whether we want to keep the resources here, for instance
```go
keepResources := os.Getenv("KEEP_RESOURCES")
if len(keepResources) == 0 {
deleteVM()
}
```